### PR TITLE
Hw 28.1

### PR DESCRIPTION
**Pull request by MaliukDaria**

### CODE REVIEW

File: src/main/java/com/online/store/web/filters/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+public class AuthenticationFilter implements Filter {
+    public static final String USER_ID = "userId";
+    private static final Injector injector = Injector.getInstance("com.online.store");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
```
Comment (fyarkiy) [inline src/main/java/com/online/store/web/filters/AuthenticationFilter.java]: I think you can remove empty line here

File: src/main/java/com/online/store/web/filters/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if (userId == null || userService.getById(userId) == null) {
+            resp.sendRedirect(req.getContextPath() + "/users/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (fyarkiy) [inline src/main/java/com/online/store/web/filters/AuthenticationFilter.java]: empty line

File: src/main/webapp/WEB-INF/views/homePage.jsp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <style>
+        body {
+            font-family: Arial, Helvetica, sans-serif;
+            margin: 0;
+        }
+
+        /* Style the header */
+
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/homePage.jsp]: Empty line?

File: src/main/webapp/WEB-INF/views/homePage.jsp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            background-color: #ddd;
+            color: #4c4c4c;
+        }
+    </style>
+</head>
+<body>
+
+<%--<div class="header">
+    <h1>Best online shop</h1>
+</div>--%>
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/homePage.jsp]: Do you need those comments lines 58-60?

File: src/main/webapp/WEB-INF/views/orders/allOrders.jsp [hunk 5]
```diff
@@ -1,11 +1,31 @@
...
+            width: 100%;
+            border: 1px solid #ddd;
+        }
+
+        th, td {
+            text-align: left;
+            padding: 16px;
+        }
+
+        tr:nth-child(even) {
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/orders/allOrders.jsp:20]: I think you can remove this empty line

File: src/main/webapp/WEB-INF/views/orders/allOrders.jsp [hunk 6]
```diff
@@ -1,11 +1,31 @@
...
+            border-spacing: 0;
+            width: 100%;
+            border: 1px solid #ddd;
+        }
+
+        th, td {
+            text-align: left;
+            padding: 16px;
+        }
+
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/orders/allOrders.jsp]: I think you can remove this empty line

File: src/main/webapp/WEB-INF/views/orders/orderDetails.jsp [hunk 7]
```diff
@@ -1,11 +1,31 @@
...
+            border-spacing: 0;
+            width: 100%;
+            border: 1px solid #ddd;
+        }
+
+        th, td {
+            text-align: left;
+            padding: 16px;
+        }
+
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/orders/orderDetails.jsp]: I think you can remove this empty line:)

File: src/main/webapp/WEB-INF/views/orders/userOrders.jsp [hunk 8]
```diff
@@ -1,11 +1,31 @@
...
+            width: 100%;
+            border: 1px solid #ddd;
+        }
+
+        th, td {
+            text-align: left;
+            padding: 16px;
+        }
+
+        tr:nth-child(even) {
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/orders/userOrders.jsp:18]: I think you can remove this empty line

File: src/main/webapp/WEB-INF/views/orders/userOrders.jsp [hunk 9]
```diff
@@ -1,11 +1,31 @@
...
     <title>Your orders</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <style>
+        table {
+            border-collapse: collapse;
+            border-spacing: 0;
+            width: 100%;
+            border: 1px solid #ddd;
+        }
+
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/orders/userOrders.jsp]: I think you can remove this empty line

File: src/main/webapp/WEB-INF/views/users/login.jsp [hunk 10]
```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <style>
+        body {
+            font-family: Arial, Helvetica, sans-serif;
+        }
+
+
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/users/login.jsp]: I think you can remove this empty line

File: src/main/webapp/WEB-INF/views/users/registration.jsp [hunk 11]
```diff
@@ -1,17 +1,91 @@
...
+        <h4 ></h4>
+        Repeat Password
+        <input type="password" required minlength="4" placeholder="Repeat Password" name="repeatPassword" required>
+        <h4 style="color:red"> ${errorMessage} </h4>
+        <button type="submit" class="registerbtn">Register</button>
+    </div>
 </form>
 </body>
-</html>
+</html>
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/users/registration.jsp]: no empty line in the end of file

**Review by fyarkiy — APPROVED** [review]

File: src/main/java/com/online/store/web/filters/AuthenticationFilter.java (hunk 1 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/web/filters/AuthenticationFilter.java]: done

File: src/main/java/com/online/store/web/filters/AuthenticationFilter.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/web/filters/AuthenticationFilter.java]: done

File: src/main/webapp/WEB-INF/views/homePage.jsp (hunk 3 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/homePage.jsp]: done

File: src/main/webapp/WEB-INF/views/homePage.jsp (hunk 4 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/homePage.jsp]: done

File: src/main/webapp/WEB-INF/views/orders/orderDetails.jsp (hunk 7 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/orders/orderDetails.jsp]: done

File: src/main/webapp/WEB-INF/views/orders/userOrders.jsp (hunk 8 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/orders/userOrders.jsp:18]: done

File: src/main/webapp/WEB-INF/views/users/login.jsp (hunk 10 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/users/login.jsp]: done

File: src/main/webapp/WEB-INF/views/users/registration.jsp (hunk 11 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/users/registration.jsp]: done

File: src/main/webapp/WEB-INF/views/orders/userOrders.jsp (hunk 9 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/orders/userOrders.jsp]: done

File: src/main/webapp/WEB-INF/views/orders/allOrders.jsp (hunk 6 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/views/orders/allOrders.jsp]: done

File: src/main/java/com/online/store/controllers/user/controllers/LoginController.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        try {
+            User user = authenticationService.login(login,password);
+            HttpSession session = req.getSession();
+            session.setAttribute("userId", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/online/store/controllers/user/controllers/LoginController.java]: let's make userId a constant

File: src/main/java/com/online/store/security/AuthenticationServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.getByLogin(login)
+                .orElseThrow(() -> new AuthenticationException("Incorrect login or password"));
+        if (user.getPassword().equals(password)) {
+            return user;
+        } else {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/online/store/security/AuthenticationServiceImpl.java]: do we need else after return?

File: src/main/java/com/online/store/security/AuthenticationServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+            throw new AuthenticationException("Incorrect login or password");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/online/store/security/AuthenticationServiceImpl.java]: either make both checks in one if() or make exception message a constant

File: src/main/java/com/online/store/controllers/user/controllers/LoginController.java (hunk 12 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/user/controllers/LoginController.java]: done

File: src/main/java/com/online/store/security/AuthenticationServiceImpl.java (hunk 13 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/security/AuthenticationServiceImpl.java]: no, removed)

File: src/main/java/com/online/store/security/AuthenticationServiceImpl.java (hunk 14 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/security/AuthenticationServiceImpl.java]: done

File: src/main/java/com/online/store/security/AuthenticationServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.getByLogin(login).orElse(null);
+        if (user != null && user.getPassword().equals(password)) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/online/store/security/AuthenticationServiceImpl.java]: let's use isPresent()

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/com/online/store/security/AuthenticationServiceImpl.java (hunk 15 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/security/AuthenticationServiceImpl.java]: done

**Pull request merged**